src/Microsoft.OData.Client/ALinq/ExpressionWriter.cs

```diff
@@ -506,7 +506,7 @@ internal override Expression VisitConstant(ConstantExpression c)
                 MemberInfo member = typeAnnotation.ElementType.GetField(c.Value.ToString());
                 string memberValue = ClientTypeUtil.GetServerDefinedName(member);
                 ODataEnumValue enumValue = new ODataEnumValue(memberValue, typeNameInEdm ?? typeAnnotation.ElementTypeName);
-                result = ODataUriUtils.ConvertToUriLiteral(enumValue, CommonUtil.ConvertToODataVersion(this.uriVersion), null);
+                result = ODataUriUtils.ConvertToUriLiteral(enumValue, CommonUtil.ConvertToODataVersion(this.uriVersion), null as IEdmModel);

```

Comment:
could also use [named parameters](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments#named-arguments) to resolve the ambiguity over which null you mean; something like 
```ConvertToUriLiteral(enumValue, CommonUtil.ConvertToODataVersion(this.uriVersion), model: null)```

Reply:
Noted. changing IsIeee754Compatible parameter to bool fixes the problem.

Reply:
Using `bool `fixes this.